### PR TITLE
Commerce 5661 define diagram model

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://issues.liferay.com/browse/COMMERCE-5661

### CODE REVIEW

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="number" type="int" />
+		<column name="CPInstanceUuid" type="String" />
+		<column name="CProductId" type="long" />
+		<column name="CPDefinitionId" type="long" />
+		<column name="positionX" type="double" />
+		<column name="positionY" type="double" />
```
Comment (dsitu) [inline modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml:54]: sort these ids up
```suggestion
		<column name="CPInstanceUuid" type="String" />
		<column name="CProductId" type="long" />
		<column name="CPDefinitionId" type="long" />
		<column name="number" type="int" />
		<column name="positionX" type="double" />
		<column name="positionY" type="double" />
```

**Review by dsitu — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml]: thanks @dsitu !

**Review by dsitu — APPROVED** [review]

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+		<column name="companyId" type="long" />
+		<column name="userId" type="long" />
+		<column name="userName" type="String" />
+		<column name="createDate" type="Date" />
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="CPInstanceUuid" type="String" />
```
Comment (andrea-ale-sbarra) [inline modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml:49]: use className classPK?

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		<column name="userId" type="long" />
+		<column name="userName" type="String" />
+		<column name="createDate" type="Date" />
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="CPDefinitionId" type="long" />
+		<column name="radius" type="double" />
+		<column name="color" type="String" />
```
Comment (andrea-ale-sbarra) [inline modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml:25]: add type for diagram?

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml (hunk 2 above)
Comment (andrea-ale-sbarra) [reply modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml:49]: Add Flag that say if link is SKU or Diagram

Add product code for product that is not present at catalog level

File: modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+		<!-- Other fields -->
+
+		<column name="CPInstanceUuid" type="String" />
+		<column name="CProductId" type="long" />
+		<column name="CPDefinitionId" type="long" />
+		<column name="number" type="int" />
+		<column name="positionX" type="double" />
+		<column name="positionY" type="double" />
+
```
Comment (andrea-ale-sbarra) [inline modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml:55]: miss quantity field

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Comment by andrea-ale-sbarra** [comment]
ci:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Comment by Commando18** [comment]
I will manually forward it since the error is not related to our work.

**Pull request closed without merging**